_drafts/2017-11-01-in-memory-testing.md

```diff
@@ -0,0 +1,59 @@
+---
+
+title: "In-memory testing in sensenet ECM"
+author: tusmester
+image: "../img/posts/inmemory_testing.png"
+tags: [test, database]
+
+---
+
+## Testing a complete ECM platform in memory
+In the previous version of sensenet ECM we had many tests, 1400+. They covered most of the codebase, but unfortunately only some of them were real unit tests. Most of them were integration tests, involving multiple layers of the product, and as such, in many cases they required a fully installed sensenet ECM database in place.
+
+That is not how a modern test environment should work, because of many reasons: speed, maintainability, flexibility.
```

Comment:
This snippet would be more appropriate, if you count to three only:
...because of the following reasons: ...